dateparser/date.py

```diff
@@ -363,7 +363,7 @@ def __init__(self, languages=None, locales=None, region=None, try_previous_local
         self.languages = languages
         self.locales = locales
         self.region = region
-        self.previous_locales = set()
+        self.previous_locales = dict()
```

Comment:
We will drop the Python 3.5 support, so it could be ok to leave `dict`, as it's ordered since Python 3.6.

**However**, as explained here: https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6#answer-39980744 it's just an implementation detail and it's since Python 3.7 that we can depend on this.

So I would move from `dict` to `collections.OrderedDict`. We can always change this in the future.